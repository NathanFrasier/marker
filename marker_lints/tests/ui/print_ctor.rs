// normalize-stdout-windows: "tests/ui/" -> "$$DIR/"

#[derive(Debug, Default)]
struct FieldStruct {
    a: u32,
    b: u32,
}
#[derive(Default)]
struct TupleStruct(u32, u32);
union Union {
    a: u32,
}
enum Enum {
    A,
    B(u32),
    C { f1: u32, f2: u32 },
}

fn main() {
    let _print_tuple = (1, 2, 3);
    let _print_array = [1, 2, 3];
    let _print_array = [1; 3];

    let _print_range = 11..;
    let _print_range = 1..3;
    let _print_range = ..3;
    let _print_range = ..=3;
    let _print_range = 1..=3;
    let _print_range = ..;

    let _print_ctor = FieldStruct { a: 1, b: 2 };
    let _print_ctor = FieldStruct { a: 10, ..FieldStruct::default() };

    let _print_ctor = Union { a: 8 };

    // let _ = TupleStruct(1, 2);
    let _print_ctor = TupleStruct { 0: 3, ..TupleStruct::default() };

    let _ = Enum::A;
    // let _ = Enum::B(1);
    // let _ = Enum::C { f1: 44, f2: 55 };
}