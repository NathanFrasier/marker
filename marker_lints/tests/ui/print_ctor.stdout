Tuple(
    TupleExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        elements: [
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 1,
                    suffix: None,
                },
            ),
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 2,
                    suffix: None,
                },
            ),
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 3,
                    suffix: None,
                },
            ),
        ],
    },
)

Array(
    ArrayExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        elements: [
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 1,
                    suffix: None,
                },
            ),
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 2,
                    suffix: None,
                },
            ),
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 3,
                    suffix: None,
                },
            ),
        ],
        len_expr: None,
    },
)

Array(
    ArrayExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        elements: [
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 1,
                    suffix: None,
                },
            ),
        ],
        len_expr: Some(
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 3,
                    suffix: None,
                },
            ),
        ),
    },
)

Range(
    RangeExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        start: Some(
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 11,
                    suffix: None,
                },
            ),
        ),
        end: None,
        is_inclusive: false,
    },
)

Range(
    RangeExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        start: Some(
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 1,
                    suffix: None,
                },
            ),
        ),
        end: Some(
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 3,
                    suffix: None,
                },
            ),
        ),
        is_inclusive: false,
    },
)

Range(
    RangeExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        start: None,
        end: Some(
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 3,
                    suffix: None,
                },
            ),
        ),
        is_inclusive: false,
    },
)

Range(
    RangeExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        start: None,
        end: Some(
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 3,
                    suffix: None,
                },
            ),
        ),
        is_inclusive: true,
    },
)

Range(
    RangeExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        start: Some(
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 1,
                    suffix: None,
                },
            ),
        ),
        end: Some(
            IntLit(
                IntLitExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    value: 3,
                    suffix: None,
                },
            ),
        ),
        is_inclusive: true,
    },
)

Range(
    RangeExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        start: None,
        end: None,
        is_inclusive: false,
    },
)

Ctor(
    CtorExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        path: AstQPath {
            self_ty: None,
            path_ty: None,
            path: AstPath {
                segments: [
                    AstPathSegment {
                        ident: Ident {
                            name: "FieldStruct",
                            span: Span {
                                source: File(
                                    "$DIR/print_ctor.rs",
                                ),
                                start: 573,
                                end: 584,
                            },
                        },
                        generics: GenericArgs {
                            args: [],
                        },
                    },
                ],
            },
            target: Item(
                ItemId(..),
            ),
        },
        fields: [
            CtorField {
                span: SpanId(..),
                ident: Ident {
                    name: "a",
                    span: Span {
                        source: File(
                            "$DIR/print_ctor.rs",
                        ),
                        start: 587,
                        end: 588,
                    },
                },
                expr: IntLit(
                    IntLitExpr {
                        data: CommonExprData {
                            _lifetime: PhantomData<&()>,
                            id: ExprId(..),
                            span: SpanId(..),
                        },
                        value: 1,
                        suffix: None,
                    },
                ),
            },
            CtorField {
                span: SpanId(..),
                ident: Ident {
                    name: "b",
                    span: Span {
                        source: File(
                            "$DIR/print_ctor.rs",
                        ),
                        start: 593,
                        end: 594,
                    },
                },
                expr: IntLit(
                    IntLitExpr {
                        data: CommonExprData {
                            _lifetime: PhantomData<&()>,
                            id: ExprId(..),
                            span: SpanId(..),
                        },
                        value: 2,
                        suffix: None,
                    },
                ),
            },
        ],
        base: None,
    },
)

Ctor(
    CtorExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        path: AstQPath {
            self_ty: None,
            path_ty: None,
            path: AstPath {
                segments: [
                    AstPathSegment {
                        ident: Ident {
                            name: "FieldStruct",
                            span: Span {
                                source: File(
                                    "$DIR/print_ctor.rs",
                                ),
                                start: 623,
                                end: 634,
                            },
                        },
                        generics: GenericArgs {
                            args: [],
                        },
                    },
                ],
            },
            target: Item(
                ItemId(..),
            ),
        },
        fields: [
            CtorField {
                span: SpanId(..),
                ident: Ident {
                    name: "a",
                    span: Span {
                        source: File(
                            "$DIR/print_ctor.rs",
                        ),
                        start: 637,
                        end: 638,
                    },
                },
                expr: IntLit(
                    IntLitExpr {
                        data: CommonExprData {
                            _lifetime: PhantomData<&()>,
                            id: ExprId(..),
                            span: SpanId(..),
                        },
                        value: 10,
                        suffix: None,
                    },
                ),
            },
        ],
        base: Some(
            Call(
                CallExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    operand: Path(
                        PathExpr {
                            data: CommonExprData {
                                _lifetime: PhantomData<&()>,
                                id: ExprId(..),
                                span: SpanId(..),
                            },
                            path: AstQPath {
                                self_ty: None,
                                path_ty: Some(
                                    Path(
                                        PathTy {
                                            data: CommonTyData {
                                                _lifetime: PhantomData<&()>,
                                                span: Some(
                                                    SpanId(..),
                                                ),
                                                is_syntactic: true,
                                            },
                                            path: AstQPath {
                                                self_ty: None,
                                                path_ty: None,
                                                path: AstPath {
                                                    segments: [
                                                        AstPathSegment {
                                                            ident: Ident {
                                                                name: "FieldStruct",
                                                                span: Span {
                                                                    source: File(
                                                                        "$DIR/print_ctor.rs",
                                                                    ),
                                                                    start: 646,
                                                                    end: 657,
                                                                },
                                                            },
                                                            generics: GenericArgs {
                                                                args: [],
                                                            },
                                                        },
                                                    ],
                                                },
                                                target: Item(
                                                    ItemId(..),
                                                ),
                                            },
                                        },
                                    ),
                                ),
                                path: AstPath {
                                    segments: [
                                        AstPathSegment {
                                            ident: Ident {
                                                name: "FieldStruct",
                                                span: Span {
                                                    source: File(
                                                        "$DIR/print_ctor.rs",
                                                    ),
                                                    start: 646,
                                                    end: 657,
                                                },
                                            },
                                            generics: GenericArgs {
                                                args: [],
                                            },
                                        },
                                        AstPathSegment {
                                            ident: Ident {
                                                name: "default",
                                                span: Span {
                                                    source: File(
                                                        "$DIR/print_ctor.rs",
                                                    ),
                                                    start: 659,
                                                    end: 666,
                                                },
                                            },
                                            generics: GenericArgs {
                                                args: [],
                                            },
                                        },
                                    ],
                                },
                                target: Item(
                                    ItemId(..),
                                ),
                            },
                        },
                    ),
                    args: [],
                },
            ),
        ),
    },
)

Ctor(
    CtorExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        path: AstQPath {
            self_ty: None,
            path_ty: None,
            path: AstPath {
                segments: [
                    AstPathSegment {
                        ident: Ident {
                            name: "Union",
                            span: Span {
                                source: File(
                                    "$DIR/print_ctor.rs",
                                ),
                                start: 695,
                                end: 700,
                            },
                        },
                        generics: GenericArgs {
                            args: [],
                        },
                    },
                ],
            },
            target: Item(
                ItemId(..),
            ),
        },
        fields: [
            CtorField {
                span: SpanId(..),
                ident: Ident {
                    name: "a",
                    span: Span {
                        source: File(
                            "$DIR/print_ctor.rs",
                        ),
                        start: 703,
                        end: 704,
                    },
                },
                expr: IntLit(
                    IntLitExpr {
                        data: CommonExprData {
                            _lifetime: PhantomData<&()>,
                            id: ExprId(..),
                            span: SpanId(..),
                        },
                        value: 8,
                        suffix: None,
                    },
                ),
            },
        ],
        base: None,
    },
)

Ctor(
    CtorExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        path: AstQPath {
            self_ty: None,
            path_ty: None,
            path: AstPath {
                segments: [
                    AstPathSegment {
                        ident: Ident {
                            name: "TupleStruct",
                            span: Span {
                                source: File(
                                    "$DIR/print_ctor.rs",
                                ),
                                start: 768,
                                end: 779,
                            },
                        },
                        generics: GenericArgs {
                            args: [],
                        },
                    },
                ],
            },
            target: Item(
                ItemId(..),
            ),
        },
        fields: [
            CtorField {
                span: SpanId(..),
                ident: Ident {
                    name: "0",
                    span: Span {
                        source: File(
                            "$DIR/print_ctor.rs",
                        ),
                        start: 782,
                        end: 783,
                    },
                },
                expr: IntLit(
                    IntLitExpr {
                        data: CommonExprData {
                            _lifetime: PhantomData<&()>,
                            id: ExprId(..),
                            span: SpanId(..),
                        },
                        value: 3,
                        suffix: None,
                    },
                ),
            },
        ],
        base: Some(
            Call(
                CallExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    operand: Path(
                        PathExpr {
                            data: CommonExprData {
                                _lifetime: PhantomData<&()>,
                                id: ExprId(..),
                                span: SpanId(..),
                            },
                            path: AstQPath {
                                self_ty: None,
                                path_ty: Some(
                                    Path(
                                        PathTy {
                                            data: CommonTyData {
                                                _lifetime: PhantomData<&()>,
                                                span: Some(
                                                    SpanId(..),
                                                ),
                                                is_syntactic: true,
                                            },
                                            path: AstQPath {
                                                self_ty: None,
                                                path_ty: None,
                                                path: AstPath {
                                                    segments: [
                                                        AstPathSegment {
                                                            ident: Ident {
                                                                name: "TupleStruct",
                                                                span: Span {
                                                                    source: File(
                                                                        "$DIR/print_ctor.rs",
                                                                    ),
                                                                    start: 790,
                                                                    end: 801,
                                                                },
                                                            },
                                                            generics: GenericArgs {
                                                                args: [],
                                                            },
                                                        },
                                                    ],
                                                },
                                                target: Item(
                                                    ItemId(..),
                                                ),
                                            },
                                        },
                                    ),
                                ),
                                path: AstPath {
                                    segments: [
                                        AstPathSegment {
                                            ident: Ident {
                                                name: "TupleStruct",
                                                span: Span {
                                                    source: File(
                                                        "$DIR/print_ctor.rs",
                                                    ),
                                                    start: 790,
                                                    end: 801,
                                                },
                                            },
                                            generics: GenericArgs {
                                                args: [],
                                            },
                                        },
                                        AstPathSegment {
                                            ident: Ident {
                                                name: "default",
                                                span: Span {
                                                    source: File(
                                                        "$DIR/print_ctor.rs",
                                                    ),
                                                    start: 803,
                                                    end: 810,
                                                },
                                            },
                                            generics: GenericArgs {
                                                args: [],
                                            },
                                        },
                                    ],
                                },
                                target: Item(
                                    ItemId(..),
                                ),
                            },
                        },
                    ),
                    args: [],
                },
            ),
        ),
    },
)

